src/EventStore.Core/TransactionLog/LogRecords/LogRecordExtensions.cs

```diff
@@ -0,0 +1,16 @@
+using EventStore.LogCommon;
+
+namespace EventStore.Core.TransactionLog.LogRecords;
+
+public static class LogRecordExtensions {
+	public static bool IsTransactionBoundary(this ILogRecord rec) {
+		return rec.RecordType switch {
+			LogRecordType.Stream => false, // log v3 stream record
+			LogRecordType.EventType => false, // log v3 event type record
+			LogRecordType.Prepare => rec is IPrepareLogRecord prepare &&
+			                         (!prepare.Flags.HasFlag(PrepareFlags.IsCommitted) || // explicit transaction
+			                         prepare.Flags.HasFlag(PrepareFlags.TransactionEnd)), // last prepare in implicit transaction
+			_ => true
```

Comment:
@YoEight it might not be safe if we add a new log record type that's not atomic. would it be preferable to enumerate all the log record types? (the performance impact should be negligible i think since apparently the compiler builds a jump/hash table if there are enough options)